coalib/results/Diff.py

```diff
@@ -508,3 +510,69 @@ def remove(self, range):
         :param range: The range to delete.
         """
         self.replace(range, '')
+
+
+def coala_unified_diff(a, b, fromfile='', tofile='', fromfiledate='',
+                       tofiledate='', n=3, lineterm='\n'):
+    r"""
+    This code is adapted from python's difflib library:
+    https://svn.python.org/projects/python/trunk/Lib/difflib.py
+
+    Compare two sequences of lines; generate the delta as a unified diff.
+
+    Unified diffs are a compact way of showing line changes and a few
+    lines of context.  The number of context lines is set by 'n' which
+    defaults to three.
+
+    By default, the diff control lines (those with ---, +++, or @@) are
+    created with a trailing newline.  This is helpful so that inputs
+    created from file.readlines() result in diffs that are suitable for
+    file.writelines() since both the inputs and outputs have trailing
+    newlines.
+
+    For inputs that do not have trailing newlines, set the lineterm
+    argument to "" so that the output will be uniformly newline free.
+
+    The unidiff format normally has a header for filenames and modification
+    times.  Any or all of these may be specified using strings for
+    'fromfile', 'tofile', 'fromfiledate', and 'tofiledate'.
+    The modification times are normally expressed in the ISO 8601 format.
+
+    If a or b misses a newline at end of file, an additional newline is
+    generated at end of last line, and an extra warning line "\ No newline
+    at end of file" is generated if warn_no_newline_at_end is True, to
+    make the patch program happy.
+    """
+
+    no_newline_at_end_warning = r'\ No newline at end of file'
+    started = False
+    for group in difflib.SequenceMatcher(None, a, b).get_grouped_opcodes(n):
+        if not started:
+            fromdate = '\t%s' % fromfiledate if fromfiledate else ''
+            todate = '\t%s' % tofiledate if tofiledate else ''
+            yield '--- %s%s%s' % (fromfile, fromdate, lineterm)
+            yield '+++ %s%s%s' % (tofile, todate, lineterm)
+            started = True
+        i1, i2, j1, j2 = group[0][
+            1], group[-1][2], group[0][3], group[-1][4]
+        yield '@@ -%d,%d +%d,%d @@%s' % (i1 + 1, i2 - i1,
+                                         j1 + 1, j2 - j1, lineterm)
+        for tag, i1, i2, j1, j2 in group:
```

Comment:
In this for loop I have introduced `if not line.endswith(lineterm):` checking and corresponding action. @Makman2